src/dpp/events/channel_delete.cpp

```diff
@@ -40,20 +38,17 @@ using namespace dpp;
  */
 void channel_delete::handle(discord_client* client, json &j, const std::string &raw) {
 	json& d = j["d"];
-	dpp::guild* g = nullptr;
-	dpp::channel* c = dpp::find_channel(snowflake_not_null(&d, "id"));
-	if (c) {
-		g = dpp::find_guild(c->guild_id);
-		if (g) {
-			auto gc = std::find(g->channels.begin(), g->channels.end(), c->id);
-			if (gc != g->channels.end()) {
-				g->channels.erase(gc);
-			}
-		}
-		dpp::get_channel_cache()->remove(c);
+	channel c = channel().fill_from_json(&d);
```

Comment:
`const`!

Reply:
no, none of these are const in the structs they go into

Reply:
ah right :)

Reply:
...actually doesnt matter, theyre assigned (not as ptrs) in the struct below